bitcoin/src/sighash.rs

```diff
@@ -1378,74 +1376,148 @@ mod tests {
         assert_eq!(expected, hash.into_inner());
     }
 
+    #[cfg(feature = "serde")]
     #[test]
     fn bip_341_sighash_tests() {
-        let data = bip_341_read_json();
-        assert!(data["version"].as_u64().unwrap() == 1u64);
-        let secp = &secp256k1::Secp256k1::new();
-        let key_path = &data["keyPathSpending"].as_array().unwrap()[0];
-
-        let raw_unsigned_tx =
-            hex_decode!(Transaction, key_path["given"]["rawUnsignedTx"].as_str().unwrap());
-        let mut utxos = vec![];
-        for utxo in key_path["given"]["utxosSpent"].as_array().unwrap() {
-            let spk = hex_script!(utxo["scriptPubKey"].as_str().unwrap());
-            let amt = utxo["amountSats"].as_u64().unwrap();
-            utxos.push(TxOut { value: amt, script_pubkey: spk });
+        fn sighash_deser_numeric<'de, D>(deserializer: D) -> Result<SchnorrSighashType, D::Error> where D: actual_serde::Deserializer<'de> {
+            use actual_serde::de::{Deserialize, Error, Unexpected};
```

Comment:
nit: Perhaps just plain `use serde::de`

Reply:
I think `serde` didn't work for some reason, I can check later but would rather not force re-review.

Reply:
I checked before I posted but I agree don't force re-review.